src/Compute/custom/NewAzVM_SimpleParameterSet.cs

```diff
@@ -0,0 +1,346 @@
+                    identity: _cmdlet.GetVMIdentityFromArgs(),
+                    proximityPlacementGroup: ppgSubResourceFunc);
+            }
+        }
+
+        async Task StrategyExecuteCmdletAsync(IAsyncCmdlet asyncCmdlet)
+        {
+            var client = new Client(this, asyncCmdlet);
+
+            ResourceGroupName = ResourceGroupName ?? Name;
+            VirtualNetworkName = VirtualNetworkName ?? Name;
+            SubnetName = SubnetName ?? Name;
+            PublicIpAddressName = PublicIpAddressName ?? Name;
+            SecurityGroupName = SecurityGroupName ?? Name;
+
+            var resourceClient = AzureSession.Instance.ClientFactory.CreateArmClient<ResourceManagementClient>(
+                    DefaultProfile.DefaultContext,
+                    AzureEnvironment.Endpoint.ResourceManager);
+
+            var parameters = new Parameters(this, client, resourceClient);
+
+
+
+            var result = await client.RunAsync(SubscriptionId, parameters, asyncCmdlet);
+
+            if (result != null)
+            {
+                var fqdn = PublicIPAddressStrategy.Fqdn(DomainNameLabel, Location);
+                var psResult = result;
+                psResult.OSProfileComputerName = fqdn;
+                var connectionString = parameters.ImageAndOsType.GetConnectionString(
+                    fqdn,
+                    Credential?.UserName);
+                asyncCmdlet.WriteVerbose(
+                    Microsoft.Azure.PowerShell.Cmdlets.Compute.Resources.VirtualMachineUseConnectionString,
+                    connectionString);
+                asyncCmdlet.WriteObject(psResult);
+            }
+        }
+
+
+        /// <summary>
+        /// Heres whats happening here :
+        /// If "SystemAssignedIdentity" and "UserAssignedIdentity" are both present we set the type of identity to be SystemAssignedUsrAssigned and set the user 
+        /// defined identity in the VM identity object.
+        /// If only "SystemAssignedIdentity" is present, we just set the type of the Identity to "SystemAssigned" and no identity ids are set as its created by Azure
+        /// If only "UserAssignedIdentity" is present, we set the type of the Identity to be "UserAssigned" and set the Identity in the VM identity object.
+        /// If neither is present, we return a null.
+        /// </summary>
+        /// <returns>Returning the Identity generated form the cmdlet parameters "SystemAssignedIdentity" and "UserAssignedIdentity"</returns>
+        private VirtualMachineIdentity GetVMIdentityFromArgs()
+        {
+            VirtualMachineIdentity identity = null;
+
+            if (this.IsParameterBound(c => c.SystemAssignedIdentity))
+            {
+                identity = new VirtualMachineIdentity();
+                identity.Type = Microsoft.Azure.PowerShell.Cmdlets.Compute.Support.ResourceIdentityType.SystemAssigned;
+            }
+
+            // ToDO: Add support for User-Assigned identities
```

Comment:
@markcowl is this something we want to do for the first preview? If not, then I think we should remove the `-UseAssignedIdentity` parameter unless we're looking to not break scripts (at the cost of not respecting this parameter 😁)

Reply:
Correct, we are looking to not break scripts - the current cmdlets actually do not do this correctly for user-assigned identities, and we need to add some support for looking up identity information to support this.